autopush-common/src/db/bigtable/bigtable_client/row.rs

```diff
@@ -15,15 +15,14 @@ pub struct Row {
 
 impl Row {
     /// Return all cells for a given column
-    pub fn get_cells(&self, column: &str) -> Option<Vec<Cell>> {
-        self.cells.get(column).cloned()
+    pub fn get_cells(&mut self, column: &str) -> Option<Vec<Cell>> {
```

Comment:
I think these methods should be renamed to `remove_cells/cell` to be clearer to the caller

Reply:
Perhaps, but the function is to return the content of the cell, not just delete them.

Reply:
But it can be a leaky abstraction potentially requiring the caller to care: because `get` method suggests it could be called repeatedly w/ the same result (idempotent). I'd also argue it should care at this level where a significant amount of data is shuffling around, thus the ownership of it vs where it's cloned should be clear.

In fact a good example of this is the grpc generated structs, offering a `get_`  method returning a reference and a `take_` version that similarly "removes" it via `mem::replace` or `take`. Renaming these methods to `take_cells/cell` instead is probably even clearer to the caller than `remove` since it follows that same pattern, how does that sound?

Reply:
Yep, I'm fine with renaming these `take_*`